### DISCUSSION TITLE
How to programmatically control column ordering in migrations?

### DISCUSSION THREAD

**Discussion by ccpu** [mikro-orm/mikro-orm#6212] [Q&A]
Hello,

I'm looking for a way to programmatically control column ordering in generated migrations so that certain "base" columns (from BaseEntity) always appear after entity-specific columns. 

Here's a simplified example:
```typescript
export class BaseEntity {
  @Property()
  createdAt!: Date;

  @Property()
  updatedAt!: Date;

  @Property({ nullable: true })
  deletedAt?: Date;
}

@Entity({ tableName: 'project_members' })
export class ProjectMemberEntity extends BaseEntity {
  @Property()
  id!: string;

  @PrimaryKey()
  projectId!: string;

  @Property()
  roleId!: string;
}
```
I'd like to define base columns that should always appear last:

```
const baseColumns = ['created_at', 'updated_at', 'deleted_at'];
```
Current output:

```
CREATE TABLE `project_members` (
  `created_at` DATETIME NOT NULL,
  `updated_at` DATETIME NOT NULL,
  `deleted_at` DATETIME NULL,
  `id` VARCHAR(255) NOT NULL,
  `project_id` VARCHAR(255) NOT NULL,
  `role_id` VARCHAR(255) NOT NULL,
  PRIMARY KEY (`project_id`)
)
```
Desired output:

```
CREATE TABLE `project_members` (
  `id` VARCHAR(255) NOT NULL,
  `project_id` VARCHAR(255) NOT NULL,
  `role_id` VARCHAR(255) NOT NULL,
  `created_at` DATETIME NOT NULL,
  `updated_at` DATETIME NOT NULL,
  `deleted_at` DATETIME NULL,
  PRIMARY KEY (`project_id`)
)
```

I'm looking for a programmatic solution through configuration or by extending MikroORM's schema/migration generators - not by manually modifying generated migration classes. Is there a way to achieve this?

**Comment by B4nan** [comment]
Maybe you could achieve this with discovery hooks. 

https://mikro-orm.io/docs/configuration#onmetadata-hook

Not sure why you care about this, keep in mind this would break if you add more columns to your entity, they will always be added at the end.
